phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -279,6 +301,31 @@ protected RegionScanner getWrappedScanner(final ObserverContext<RegionCoprocesso
             final ImmutableBytesWritable ptr) {
         return new RegionScanner() {
 
+            private boolean hasReferences = checkForReferenceFiles();
+            private HRegionInfo regionInfo = c.getEnvironment().getRegionInfo();
+            private byte[] actualStartKey = getActualStartKey();
+
+            private boolean checkForReferenceFiles(){
```

Comment:
Added the code comments and checking references is not expensive. We need to do this for local indexes case only.